tests/test_jsonrpc.py

```diff
@@ -79,6 +79,13 @@ class MyProtocol(JSONRPC):
         MyProtocol._request_args({})
 
 
+def test_exception_is_hashable():
+    try:
+        hash(CodeMessageError(0, ''))
+    except TypeError:
+        assert False
```

Comment:
alternatively, if you think it's nicer, this would also work I think:
`assert isinstance(CodeMessageError(0, ''), collections.abc.Hashable)`

Reply:
You don't need the try ... except.  Simply the hash line alone will do.  If it throws the test will fail.

Reply:
oh.. right :P